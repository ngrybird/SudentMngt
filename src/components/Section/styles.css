.section-container {
  width: 395px;
  height: 151px;
  border: solid 0px;
  border-radius: 5px;
  position: relative;
  margin-left: 40px;
  margin-top: 20px;
  background: white;
  box-shadow: 0 8px 6px -6px grey;
}
.hover-image {
  position: absolute;
  bottom: 68px;
  left: 67px;
  z-index: 1000;
  height: 18px;
  width: 18px;
}
.section-data {
  margin-top: 40px;
  margin-left: 40px;
}
.section-title {
  display: inline-block;
  position: relative;
  bottom: 40px;
  left: 20px;
  font-size: 28px;
}
.section-subtext {
  font-size: 12px;
  color: grey;
  margin-left: 5px;
}
.section-subtitle {
  display: inline-block;
  position: absolute;
  left: 130px;
  bottom: 50px;
  font-size: 14px;
}
