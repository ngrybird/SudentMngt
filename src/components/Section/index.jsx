import React from "react";
import "./styles.css";
import Elipsis from "../../images/Ellipse.png";
import userIcon from "../../images/user.png";
import thunderIcon from "../../images/thunder.png";

class Section extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      ...this.props,
      icon: this.props.title === "Students" ? userIcon : thunderIcon
    };
  }
  render() {
    return (
      <div className="section-container" data-testid="custom-sections">
        <div className="section-data">
          <span>
            <img src={Elipsis} />
            <img src={this.state.icon} className="hover-image" />
          </span>
          <div className="section-title">
            {this.state.line1}
            {this.state.line2 && (
              <span className="section-subtext">{this.state.line2}</span>
            )}
          </div>
          <div className="section-subtitle">{this.state.title}</div>
        </div>
      </div>
    );
  }
}

export default Section;
