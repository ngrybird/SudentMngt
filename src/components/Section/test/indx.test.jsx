import React from "react";
import { render } from "@testing-library/react";
import Section from "../index";
import { sectionsData } from "../../Dashboard/test/testData";

describe("<Section/>", () => {
  it("Verify Section title and subtitle for section", () => {
    const { container, getByText } = render(<Section {...sectionsData[0]} />);

    const sectionTitle = getByText("62");
    const sectionSubtitle = getByText("Students");

    expect(sectionTitle).toBeInTheDocument();
    expect(sectionSubtitle).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
