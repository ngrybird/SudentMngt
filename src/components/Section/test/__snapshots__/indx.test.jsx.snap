// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section/> Verify Section title and subtitle for section 1`] = `
<div>
  <div
    class="section-container"
    data-testid="custom-sections"
  >
    <div
      class="section-data"
    >
      <span>
        <img
          src="Ellipse.png"
        />
        <img
          class="hover-image"
          src="user.png"
        />
      </span>
      <div
        class="section-title"
      >
        62
      </div>
      <div
        class="section-subtitle"
      >
        Students
      </div>
    </div>
  </div>
</div>
`;
